sessionctx/variable/setvar_affect.go

```diff
@@ -0,0 +1,116 @@
+	"tidb_index_join_batch_size":                       {},
+	"tidb_index_lookup_size":                           {},
+	"tidb_index_serial_scan_concurrency":               {},
+	"tidb_allow_batch_cop":                             {},
+	"tidb_allow_mpp":                                   {},
+	"tidb_enforce_mpp":                                 {},
+	"tidb_max_bytes_before_tiflash_external_join":      {},
+	"tidb_max_bytes_before_tiflash_external_group_by":  {},
+	"tidb_max_bytes_before_tiflash_external_sort":      {},
+	"tidb_min_paging_size":                             {},
+	"tidb_max_paging_size":                             {},
+	"tidb_enable_cascades_planner":                     {},
+	"tidb_merge_join_concurrency":                      {},
+	"tidb_index_merge_intersection_concurrency":        {},
+	"tidb_opt_projection_push_down":                    {},
+	"tidb_enable_vectorized_expression":                {},
+	"tidb_opt_join_reorder_threshold":                  {},
+	"tidb_enable_index_merge":                          {},
+	"tidb_enable_extended_stats":                       {},
+	"tidb_isolation_read_engines":                      {},
+	"tidb_executor_concurrency":                        {},
+	"tidb_partition_prune_mode":                        {},
+	"tidb_enable_index_merge_join":                     {},
+	"tidb_enable_ordered_result_mode":                  {},
+	"tidb_enable_pseudo_for_outdated_stats":            {},
+	"tidb_stats_load_sync_wait":                        {},
+	"tidb_cost_model_version":                          {},
+	"tidb_index_join_double_read_penalty_cost_rate":    {},
+	"tidb_default_string_match_selectivity":            {},
+	"tidb_enable_prepared_plan_cache":                  {},
+	"tidb_enable_non_prepared_plan_cache":              {},
+	"tidb_plan_cache_max_plan_size":                    {},
+	"tidb_opt_range_max_size":                          {},
+	"tidb_opt_advanced_join_hint":                      {},
+	"tidb_opt_prefix_index_single_scan":                {},
+	"tidb_store_batch_size":                            {},
+	"mpp_version":                                      {},
+	"tidb_enable_inl_join_inner_multi_pattern":         {},
+	"tidb_opt_enable_late_materialization":             {},
+	"tidb_opt_ordering_index_selectivity_threshold":    {},
+	"tidb_opt_enable_mpp_shared_cte_execution":         {},
+	"tidb_opt_fix_control":                             {},
+	"tidb_runtime_filter_type":                         {},
+	"tidb_runtime_filter_mode":                         {},
+	// Variables that is compatible with MySQL.
+	"cte_max_recursion_depth": {},
+	"sql_mode":                {},
+	"max_execution_time":      {},
+}
+
+func setHintUpdatable(vars []*SysVar) {
+	for _, v := range vars {
+		if _, ok := isHintUpdatable[v.Name]; ok {
+			v.IsHintUpdatable = true
+			if v.Type == TypeStr || v.Type == TypeEnum || v.Type == TypeTime || v.Type == TypeDuration {
+				fmt.Println(v.Name)
```

Comment:
debug info？

Reply:
Yes, it's used for generating test cases.